src/env_node.mli

```diff
@@ -0,0 +1,19 @@
+open Import
+
+type t
+
+val make :
+  dir:Path.t ->
+  inherit_from:t Lazy.t option ->
+  scope:Scope.t ->
+  config:Dune_env.Stanza.t ->
+  t
+
+val ocaml_flags :
+  t ->
+  profile:string ->
+  eval:(scope:Scope.t ->
+        dir:Path.t ->
+        Ordered_set_lang.Unexpanded.t ->
+        standard:(unit, string list) Build.t -> (unit, string list) Build.t) ->
```

Comment:
Tbh, I'd be against this PR because introducing types like this doesn't really make the code any clearer or well factored. I think the situation can be improved if we move all the expansion stuff to its own module as well.

Reply:
I agree that it's not great. I'll try to find a better seam, otherwise this will have to wait until expansion is extracted.